src/lib/hooks/use-wallet.ts

```diff
@@ -38,7 +38,7 @@ export const useWallet = (): Account => {
     : null
   const isConnected = !!address
   return {
-    address,
+    address: address as `0x${string}` | undefined,
```

Comment:
This is because, this could be `undefined` (and wasn't typed correctly) or we want this to be undefined in some cases?

Reply:
The earlier version of `viem` that we were using returned the type as `0x${string} | undefined` instead of `string | undefined`, which didn't match the type on `Account`. On second look, I think updating `Account` to represent the new `address` type is cleaner, so updating that instead given the new type returned by viem.